README.rst

```diff
@@ -22,14 +32,117 @@ float32     Signed    32    8      -3.402823466385288598117041834e+38  3.4028234
+
+Our encoding scheme works as follows:
+  * integers are first encoded in a byte array following the machine's endianness and then, if the machine is Little
+    Endian, the array is mirrored; then, the first byte (which will now be the MSB irregardless of the machine's
+    endianness) is XOR'd with 10000000, hence the sign bit is flipped.
+    For example, for 8-bit integers, the result of two's complement is
+    [10000000 (-128), 10000001 (-127), ..., 11111111 (-1), 00000000 (0), 00000001 (1), ..., 01111111 (127)].
+
+    Applying our encoding scheme will result in the ordering
+    [00000000 (-128), 00000001 (-127), ..., 01111111 (-1), 10000000 (0), 10000001 (1), ..., 11111111 (127)].
+
+  * floating-point numbers are encoded in a byte array following the machine's endianness and then, if the machine is
+    Little Endian, the array is mirrored; then, if the number is nonnegative, the first byte (which will now be the MSB
+    irregardless of the machine's endianness) is OR'd with 0x80 (binary 10000000), hence the sign bit is set to 1
+    (treating the two signed zeroes as one); otherwise, the whole byte array is negated, which results in flipping
+    all bits and thus reverse-ordering negative numbers. Plus and minus infinity can be encoded in this scheme and will
+    come after and before all other byte arrays, respectively.
+
+Usage
+=====
+
+As a Pyhton library
+-------------------
+
+To use the encoding and decoding functions, you need to invoke one of the ``numenc.from_TYPE()`` and
+``numenc.to_TYPE()`` functions, replacing TYPE with any one of the types listed above. For example:
+
+.. code-block:: python
+
+    >>> import numenc
+
+    >>> numenc.from_int32(1200)
+    b'\x80\x00\x04\xb0'
+    >>> numenc.to_int32(b'\x80\x00\x04\xb0')
+    1200
+    >>> numenc.from_int32("a string")
+    Traceback (most recent call last):
+     ...
+    TypeError: Wrong input: expected signed 32-bit integer.
+
+    # negative input throws ValueError for unsigned types
+    >>> numenc.from_uint8(-1)
+    Traceback (most recent call last):
+     ...
+    ValueError: expected 8-bit unsigned integer (range [0, 255]), got -1.
+
+    >>> numenc.from_int64(-999999)
+    b'\x7f\xff\xff\xff\xff\xf0\xbd\xc1'
+    >>> numenc.to_int64(b'\x7f\xff\xff\xff\xff\xf0\xbd\xc1')
+    -999999
+
+    # small rounding differences are possible
+    >>> numenc.from_float32(-23.44443)
+    b'>Dq\xce'
+    >>> numenc.to_float32(b'>Dq\xce')
+    -23.444429397583008
+
+    >>> numenc.from_float64(-23.44443)
+    b'?\xc8\x8e9\xd5\xe4\xa3\x82'
+    >>> numenc.to_float64(b'?\xc8\x8e9\xd5\xe4\xa3\x82')
+    -23.44443
+
+    >>> numenc.from_float32(float("-inf"))
+    b'\x00\x7f\xff\xff'
+    >>> numenc.from_float32(float("inf"))
+    b'\xff\x80\x00\x00'
+
+
+As a command line tool
+----------------------
+You can experiment with numenc on the command line by running the executable
+bin/pynumenc, which wraps the numenc library. For convenience, input and output
```

Comment:
Please strip "bin/"; pip automatically installs it into the bin folder (see https://www.reddit.com/r/learnpython/comments/8816u8/python_package_bin_folder/)